### DISCUSSION TITLE
inlabru for nlme?

### DISCUSSION THREAD

**Discussion by wdkrnls** [inlabru-org/inlabru#239] [General]
Hi, I read a paper suggesting iterated INLA would be very competitive for Physics/Engineering system modeling. More, humbly, I am wondering whether anyone has looked at using nonlinear mixed effects models with inlabru? Atleast the sort of models which you can fit with nlme::nlme.

**Comment by finnlindgren** [comment]
I tried to find out precisely what models nlme handles, but it's not entirely clear to me what the estimation method is; in seemed to me it doesn't do generalized additive models, and only does nonlinear least squares, with non-linear link functions? In that case INLA (at least with inlabru) can handle a wider range of models. Link function nonlinearity is handled by INLA directly (for the standard range of link functions), and inlabru can handle some additional nonlinearity in the predictor expression itself. If you have a specific example in mind, I may give a more specific answer.

**Reply by wdkrnls** [reply]
Yes, nlme::nlme just sticks to Gaussian mixed effects models with nonlinear predictor expressions. It helps you explore separate sets of linear predictors for each of the nonlinear parameters, but it also provides a lot more flexibility than lme4::nlmer or rstanarm::stan_nlmer for exploring complex heterogeneous variance structures, grouping correlation structures, and within-group residual correlation error structures within that limited scope. These are very relevant for longitudinal data generated from controlled experiments. 

The nlme package and the book by Pinhiero and Bates gives many examples for fitting these models, but it's not available for free online. However, the original paper is: ftp://www.biostat.wisc.edu/pub/lindstrom/papers/biometrics.1990.pdf

The nlraa package has some vignettes which demonstrate the usage of nlme pretty well and provides much of the bootstrapping and model averaging infrastructure needed to actually use these models in practice.

https://cran.r-project.org/web/packages/nlraa/vignettes/nlraa-AgronJ-paper.html

I would love to see how to model the data in that vignette with inlabru. Relying on nlme has a lot of issues with initialize starting parameters for the optimization process, which I'm hopeful iterated INLA could dispense with. If it can, then the next step is handling systems of ODEs a la RxODE as is done with nlmixr2, which is a wrapper for using nlme with RxODE along with two other probably more reliable but seemingly less flexible estimation methods.
